xbmc/video/dialogs/GUIDialogSubtitles.cpp

```diff
@@ -577,6 +579,11 @@ void CGUIDialogSubtitles::OnDownloadComplete(const CFileItemList *items, const s
 
   // Close the window
   Close();
+  
+  // Close subtitle OSD settings window if open
+  CGUIDialog *pDialog = CServiceBroker::GetGUI()->GetWindowManager().GetDialog(WINDOW_DIALOG_SUBTITLE_OSD_SETTINGS);
```

Comment:
a modal dialog is not supposed to close its parent. you should notifiy the parent about the event.